src/components/radio-button/radio-button.component.tsx

```diff
@@ -138,32 +141,83 @@ export const RadioButton = React.forwardRef<
       ...props,
     };
 
+    const commonPropsForNewValidation = {
```

Comment:
comment: there's a lot of duplication here as most of the props are needed whether the flag is set or not

Reply:
Suggestion: you could remove the duplication by doing something like the below but happy for you to go with what you prefer as it's not a blocking issue for me

``` 
const validationProps = {
      disabled,
      inputWidth,
      error,
      warning,
      info,
    };

    const commonProps = {
      ...validationProps,
      fieldHelpInline,
      labelSpacing,
    };

    const inputProps = {
      ...(validationRedesignOptIn ? {...commonProps, fieldHelp} : validationProps),
      autoFocus,
      checked,
      fieldHelp,
      name,
      onChange: handleChange,
      onBlur,
      onFocus,
      labelAlign,
      labelInline: true,
      labelWidth,
      label,
      labelHelp,
      id,
      value,
      type: "radio",
      /**
       * Invert the reverse prop, to ensure the FormField component renders the components
       * in the desired order (other elements which use FormField render their sub-components the
       * opposite way around by default)
       */
      reverse: !reverse,
      required,
      ref: ref || inputRef,
      ...props,
    };

    invariant(
      !props.children,
      "This component is meant to be used as a self-closing tag. " +
        "You should probably use the label prop instead."
    );

    const componentToRender = (
      <RadioButtonStyle
            applyNewValidation={validationRedesignOptIn}
            data-component={dataComponent}
            data-role={dataRole}
            data-element={dataElement}
            inline={inline}
            reverse={reverse}
            size={size}
            {...(validationRedesignOptIn ? {...commonProps, fieldHelp} : validationProps)}
            {...marginProps}
          >
            <CheckableInput {...inputProps}>
              <RadioButtonSvg />
            </CheckableInput>
          </RadioButtonStyle>
    )

    return (
      <>
        {validationRedesignOptIn ? (
          componentToRender
        ) : (
          <TooltipProvider
            helpAriaLabel={helpAriaLabel}
            tooltipPosition={tooltipPosition}
          >
            {componentToRender}
          </TooltipProvider>
        )}
      </>
    );
    ```

Reply:
Yeah will be neater way of doing it 👍🏻